File: src/main/java/io/zeitwert/fm/building/service/api/BuildingEvaluationService.java
package io.zeitwert.fm.building.service.api;

import io.zeitwert.fm.building.model.ObjBuilding;
import io.zeitwert.fm.building.service.api.dto.BuildingEvaluationResult;

public interface BuildingEvaluationService {

	/**
	 * Get the accumulated cost projection for a given building
	 * 
	 * @param building the building
	 * @return building evaluation
	 */
	BuildingEvaluationResult getEvaluation(ObjBuilding building);

}

File: fm-domain/src/main/java/io/zeitwert/fm/building/api/BuildingEvaluationService.kt
package io.zeitwert.fm.building.api

import io.zeitwert.fm.building.api.dto.BuildingEvaluationResult
import io.zeitwert.fm.building.model.ObjBuilding

interface BuildingEvaluationService {

	/**
	 * Get the accumulated cost projection for a given building
	 *
	 * @param building the building
	 * @return building evaluation
	 */
	fun getEvaluation(building: ObjBuilding): BuildingEvaluationResult

}
